### PR TITLE
Bug fix/annotations list

### PR DESCRIPTION
**Pull request by tmarkn**
- added json uploading along side txt
- added alerts for invalid files
- moved deletelist out of tagModel to app.js (wasn't needed)
- redid annotation list (correctly updates with label rename and deletion) (also added colors)
- fixed a bug with sorcery (no matches throws an error)
- fixed a bug with label deletion (deleting a label wouldn't show which new label was selected)

### CODE REVIEW

**Review by ghost — DISMISSED** [review]

**Comment by keithudock** [comment]
can you fix the merge conflict?

**Review by ghost — DISMISSED** [review]

**Review by keithudock — DISMISSED** [review]
noice

**Review by ghost — APPROVED** [review]

File: static/js/app.js [hunk 1]
```diff
@@ -114,11 +136,12 @@ textArea.on('mouseup', function (e) {
 textArea.on('contextmenu', function (e) {
   event.preventDefault();
   let position = textArea[0].selectionStart;
-  let annotations = tagModel.getAnnotationsAtPos(position);
-  if (annotations.length > 0) {
+  deleteList = tagModel.currentDoc.getAnnotationsAtPos(position);
+
+  if (deleteList.length > 0) {
     $('#delete-menu').append('<h6>Delete Annotation:</h6><hr style="margin: 0;">')
```
Comment (ghost) [inline static/js/app.js]: #delete-menu is duplicated a lot, you should store it in a variable

File: static/js/app.js [hunk 2]
```diff
@@ -158,7 +180,7 @@ $('#add-label').on('click', function () {
 });
 
 //change the document's label context
-$('#label-list').on('click', '.label', function () {
+$('#label-list').on('mouseup', '.label', function () {
```
Comment (ghost) [inline static/js/app.js]: I created a variable for this already, you should refactor a little to use it

**Review by ghost — CHANGES_REQUESTED** [review]

File: static/js/app.js (hunk 1 above)
Comment (tmarkn) [reply static/js/app.js]: updated

File: static/js/app.js (hunk 2 above)
Comment (tmarkn) [reply static/js/app.js]: updated

**Review by ghost — APPROVED** [review]

**Pull request merged**